apps/cowswap-frontend/src/legacy/state/orders/middleware/appziMiddleware.ts

```diff
@@ -58,12 +97,24 @@ function _triggerNps(
   const isLimitOrderRecentlyTraded =
     uiOrderType === UiOrderType.LIMIT && npsParams?.traded && isOrderInPendingTooLong(openSince)
 
-  // Do not show NPS if the order is hidden and expired
-  const isHiddenAndExpired = order?.isHidden && npsParams?.expired
+  // Do not show NPS if the order is hidden
+  const isHidden = order?.isHidden
```

Comment:
Never trigger NPS when the order is hidden.
This is the case for SC orders that were not signed.